server/src/test/java/org/elasticsearch/transport/OutboundHandlerTests.java

```diff
@@ -119,7 +122,12 @@ public void testSendRequest() throws IOException {
         String action = "handshake";
         long requestId = randomLongBetween(0, 300);
         boolean isHandshake = randomBoolean();
-        boolean compress = randomBoolean();
+        boolean compress;
+        if (compressionScheme == Compression.Scheme.LZ4 && version.before(Compression.Scheme.LZ4_VERSION)) {
+            compress = false;
```

Comment:
This compress boolean is used both in `handler.sendRequest` and for validation afterwards. I think we should split it into two so that we can verify the fallback to not compress against old versions.